### PR TITLE
Clean up router mechanism

### PR DESCRIPTION
**Pull request by henghonglee**
This makes bot router much more reusable for other bots in the future

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -18,7 +21,14 @@ app.use(
 )
 app.use(bodyParser.json())
 
-app.post('/thanksbot', requestProcessor.process_request);
+app.post('/thanksbot', (req, res, next) => {
```
Comment (henghonglee) [inline server.js:24]: more bots can be added here next time

File: app/bot-router.js [hunk 2]
```diff
@@ -1,43 +1,31 @@
...
-export function process_request(req, res) {
+export function processRequest(req, res, response_chain) {
+  console.log(req.body)
   var body = req.body;
-  console.log(body)
-
-  // Request params
-  const channel = body && body.event && body.event.channel
-  if (body.challenge && !channel) { // handle challenge
+  if (body.challenge) { // handle challenge
```
Comment (baimao8437) [inline app/bot-router.js:9]: actually idk what is challenge for @@

Comment (henghonglee) [reply app/bot-router.js:9]: challenge is for the verify endpoint on slack

Comment (henghonglee) [reply app/bot-router.js:9]: oh yeah i probably can move it out

Comment (henghonglee) [reply app/bot-router.js:9]: nvm can just leave it

File: server.js (hunk 1 above)
Comment (baimao8437) [reply server.js:24]: nice

**Pull request merged**